crypto/fipsmodule/cipher/internal.h

```diff
@@ -170,6 +175,30 @@ void AES_cfb1_encrypt(const uint8_t *in, uint8_t *out, size_t bits,
 void AES_cfb8_encrypt(const uint8_t *in, uint8_t *out, size_t len,
                       const AES_KEY *key, uint8_t *ivec, int *num, int enc);
 
+
+// EXPERIMENTAL functions for use in the TLS Transfer function. See |SSL_to_bytes| for more details.
+
+//  EVP_AEAD_CTX_serialize_state serializes the state of |ctx|,
+// and writes it to |cbb|. The serialized bytes contains only the subset of data
+// necessary to restore the state of an |EVP_AEAD_CTX| after initializing a new
+// instance using |EVP_AEAD_CTX_init|. If the given |EVP_AEAD| algorithm does
+// not have a state serialization function, this function will return 1, but
+// will write no content to |cbb|. Otherwise the function returns 1 on success
+// or zero for an error.
+OPENSSL_EXPORT int EVP_AEAD_CTX_serialize_state(const EVP_AEAD_CTX *ctx,
+                                                CBB *cbb);
+
+// **EXPERIMENTAL** EVP_AEAD_CTX_deserialize_state deserializes the state
+// contained in |cbs|, configures the |ctx| to match. The deserialized bytes
+// contains only the subset of data necessary to restore the state of an
+// |EVP_AEAD_CTX| after initializing a new instance using |EVP_AEAD_CTX_init|.
+// If the given |EVP_AEAD| algorithm does not have a state deserialization
+// function, this function will return 1 iff the |cbs| length is zero, otherwise
```

Comment:
The scenario I am thinking of here is that if a user provides a CBS whose length > 0 and we don't have a de-serialization function associated with that EVP_AEAD, then that should be an error. A success implies that we were able to deserialize state from the CBS which would not be the case. How did a user get a non-zero length serialization for a cipher if it doesn't support serialization/deserialization? So just avoid the issue where a user thinks they have a state to deserialize from but we can't actually restore state since it is not supported for the given EVP_AEAD. That could be because they rolled back the software version etc, but we already acknowledged in earlier design reviews that forward compatibility is not a scenario we are try to solve with this feature right now.